### DISCUSSION TITLE
[Feature] Retrieve tables and column description from big query

### DISCUSSION THREAD

**Discussion by Rkejji** [dbt-labs/dbt-bigquery#816] [Ideas]
**Description:**

I am building a data pipeline from large source raw tables, with hundreds of columns. 
I describe my sources in a yaml file in dbt using the package [codgen](https://github.com/dbt-labs/dbt-codegen). 
The problem with that package is that it only generates empty description placeholders but not the actual tables and columns descriptions.
That is normal because the description is not retrieved by dbt it is not an attribute of the [BigqueryColumn](https://github.com/dbt-labs/dbt-bigquery/blob/main/dbt/adapters/bigquery/column.py#L11C1-L11C1) class.

My idea is that when calling the bigquery API [here](https://github.com/dbt-labs/dbt-bigquery/blob/main/dbt/adapters/bigquery/column.py#L11C1-L11C1) with the function [get_table](https://cloud.google.com/python/docs/reference/bigquery/latest/google.cloud.bigquery.client.Client#google_cloud_bigquery_client_Client_get_table) from bigquery client API. We retrieve the description from the returned [table object](https://cloud.google.com/python/docs/reference/bigquery/latest/google.cloud.bigquery.table.Table) such as the table description and the [columns description](https://cloud.google.com/python/docs/reference/bigquery/latest/google.cloud.bigquery.schema.SchemaField). This will make the column and table description available in the Relation class to be used by the codgen package.

**Why is this feature important ?**

When dealing with large source data such as tables from SAP, copying manually hundreds of descriptions can be tedious, using codgen package helps but it doesn't generate descriptions.

What do you think about it ?

**Comment by K-Oxon** [comment]
I agree with your idea of importing BigQuery descriptions into dbt YAML files!
If the adapter can fetch this description, it would benefit many use cases like codegen or dbt-osmosis.

**Comment by henry-yu-gce** [comment]
I agree, and I would be super appreciative of such a feature. 
There's another similar train of thought using macros [link](https://discourse.getdbt.com/t/how-to-extract-table-comments-in-the-db-and-populate-schema-description-with-it/11619).
For BigQuery specifically, grabbing the column / table descriptions from INFORMATION_SCHEMA could be alternative solution.

**Comment by Rkejji** [comment]
I created the following issue: https://github.com/dbt-labs/dbt-bigquery/issues/1458
